ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,536 @@
+  for (i = 0; i < dims; i++)
+    dim[i] = shapes[dims - i - 1];
+
+  /* fill remaining entries with 0 */
+  for (i = dims; i < NNS_TENSOR_RANK_LIMIT; ++i) {
+    dim[i] = 0;
+  }
+
+  return 0;
+}
+
+/**
+ * @brief return the type of tensor
+ * @return 0 if OK. non-zero if error.
+ */
+int
+onnxruntime_subplugin::getTensorType (ONNXTensorElementDataType _type, tensor_type *type)
+{
+  tensor_type res;
+  *type = _NNS_END;
+
+  switch (_type) {
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8:
+      res = _NNS_INT8;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8:
+      res = _NNS_UINT8;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT16:
+      res = _NNS_INT16;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT16:
+      res = _NNS_UINT16;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32:
+      res = _NNS_INT32;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT32:
+      res = _NNS_UINT32;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64:
+      res = _NNS_INT64;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64:
+      res = _NNS_UINT64;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT:
+      res = _NNS_FLOAT32;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE:
+      res = _NNS_FLOAT64;
+      break;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16:
+#ifdef FLOAT16_SUPPORT
+      res = _NNS_FLOAT16;
+      break;
+#endif
+    default:
+      nns_logw ("Tensor type not supported: %d", (gint) _type);
+      return -EINVAL;
+  }
+
+  *type = res;
+  return 0;
+}
+
+/**
+ * @brief Method to get empty object.
+ */
+tensor_filter_subplugin &
+onnxruntime_subplugin::getEmptyInstance ()
+{
+  return *(new onnxruntime_subplugin ());
+}
+
+/**
+ * @brief Method to prepare/configure onnxruntime instance.
+ */
+void
+onnxruntime_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  Ort::AllocatorWithDefaultOptions allocator;
+
+  if (!empty_model) {
+    /* Already opened */
+
+    if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+      std::cerr << "Model path is not given." << std::endl;
+      throw std::runtime_error ("Model path is not given.");
+    }
+
+    cleanup ();
+  }
+
+  assert (model_path == nullptr);
+
+  if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
+    const std::string err_msg
+        = "Given file " + (std::string) prop->model_files[0] + " is not valid";
+    cleanup ();
+    throw std::runtime_error (err_msg);
+  }
+
+  model_path = g_strdup (prop->model_files[0]);
+
+  /** Read a model */
+  env = Ort::Env (ORT_LOGGING_LEVEL_WARNING, "nnstreamer_onnxruntime");
+  session = Ort::Session (env, model_path, sessionOptions);
+
+  input_num_tensors = session.GetInputCount ();
+  if (input_num_tensors <= 0 || input_num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    cleanup ();
+    nns_logw ("Too many input tensors: %zu max: %d", input_num_tensors, NNS_TENSOR_SIZE_LIMIT);
```

Comment:
it is ok to do nns_logw() and keep it running? Isn't this an error requiring throw?

Reply:
1. Don't use nns_logw (it's for warning).
2. Write the throw message with the specific info.
E.g.,
```C++
throw std::invalid_argument(std::string(".....") + to_string(number) + std::string("...."));
```

Reply:
Thanks. updated it.